.github/workflows/release--build-hotfix.yml

```diff
@@ -1,24 +1,50 @@
 name: "[Deployment] Create a hotfix release"
 
-on:
-  workflow_dispatch:
-    inputs:
-      releaseNumber:
-        description: "Numéro de la release à hotfix (ex: 200)"
-        required: true
-        type: string
+run-name: Create a hotfix release from ${{ github.ref }}
 
-      releaseVersion:
-        description: "Nom du hotfix (ex: 200.0.2)"
-        required: true
-        type: string
+on: workflow_dispatch
 
 jobs:
+  check-worflow-ref:
+    name: Check worflow ref
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check worflow ref
+        run: |
+          if ! [[ ${{ github.ref }} == refs/heads/maint\/\/v* ]]; then
+              echo "This workflow can only be triggered from a maintenance branch (starting with a 'maint/v')"
+              exit 1
+          fi
+
+  find-hotfix-tag-number:
+    # Find the next hotfix version available
+    # Example: given branch maint/v200 and existing tags v200.0.0, v200.0.1 then output == 200.0.2
+    name: Find hotfix tag number
+    needs: check-worflow-ref
+    runs-on: ubuntu-latest
+    outputs:
+      hotfixTagNumber: ${{ steps.find-hotfix-tag-number.outputs.hotfixTagNumber }}
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.4
+        with:
+          persist-credentials: false
+          fetch-depth: 0 # Fetch all history to get all tags
+      - id: find-hotfix-tag-number
+
+        run: |
+          GITHUB_REF=${{ github.ref }}
+          MAJOR_NUMBER=(${GITHUB_REF//refs\/heads\/maint\/v/ })
+          CURRENT_PATCH_NUMBER=$(git tag -l "v${MAJOR_NUMBER}.0.*" | sort -V | tail -n 1 | cut -d. -f3)
+          NEXT_PATCH_NUMBER=$((CURRENT_PATCH_NUMBER + 1))
+          echo "hotfixTagNumber=${MAJOR_NUMBER}.0.${NEXT_PATCH_NUMBER}" >> $GITHUB_OUTPUT
+
   build-and-tag-version:
+    needs: find-hotfix-tag-number
     name: Build and tag version
     uses: ./.github/workflows/reusable--build-and-tag.yml
     secrets: inherit
     with:
-      base_ref: maint/v${{ github.event.inputs.releaseNumber }}
-      tag_number: ${{ github.event.inputs.releaseVersion }}
+      base_ref: ${{ github.ref }}
```

Comment:
Normalement on a plus besoin de ça, il suffit que reusable--build-and-tag fasse un simple checkout non ?
Sauf que le workflow de release--build donne aussi une ref, lui aussi en fait il vaudrait mieux qu'il s'éxecute sur la ref, plutôt qu'avec la ref en argument ?
Peut être à voir plus tard pour améliorer

Reply:
oui t'as raison je pense